IPs/RSKIP176.md

```diff
@@ -0,0 +1,115 @@
+- *string* LiquidityProviderBtcAddress: Base58 string representing  the Liquidity Provider BTC refund address
+- *bool* shouldTransferToContract: Status provided by Liquidity Provider Contract to know if should transfer value to contract or not
+
+
+**Flow**:
+
+- Bridge receives a call from a contract (an alleged Liquidity Bridge Contract) and validates that the sender of that call is effectively the LBC address, that is received as parameter
+
+- Performs the same validations the Bridge does in **registerBtcTransaction** to determine if the BTC transaction is legitimate (e.g.: txAlreadyProcessed, confirmations, PMT, etc)
+
+- Verifies tx is sending funds to address derivated from Federation using the provided derivation arguments and hashing them:
+
+	`Sha256Hash(derivationArgumentsHash, userRefundAddress, LBCAddress, LPBtcAddress)`
+
+- Checks if the derivation hash was used already. If so, it will refund the user.
+
+- Verifies provided Liquidity Bridge Contract address actually belongs to a contract
+
+- Depending on the shouldTransferToContract value received as parameter from Liquidity Bridge Contract:
+
+ - If status obtained is **true**, will check if Locking Cap value is surpassed:
+    - If not surpassed:
+		- Bridge will transfer value to the Liquidity Bridge Contract
+		- will save UTXO to derived address
+		- will save UTXO in BridgeStorage (UTXO <-> Sha256Hash(Derivation Arguments))
+		- will mark tx as processed
+		- will store the UTXO for the Bridge to use, among any data concerning to the Bridge operation.
+
+	- If surpassed, will refund Liquidity Provider
+
+ - If status obtained is **false**:
+   - Refund user
+
+- Returns an integer value indicating the execution status:
+
+   ` -1: Refund user`
+
+   `-2: Refund Liquidity Provider`
+
+   `>0: Value transfered to user`
+
+
+#### Limitations
+- Transactions with depth bigger than 4320 will not be processed. That is enough depth to be able to search backwards one month worth of blocks.
+If given depth is bigger than the accepted limit, funds will be lost
+
+- If given transaction is invalid or provided address cannot be calculated from the given derivation arguments, funds will be lost
+
+
+#### Flow chart
+![registerBtcTransfer() flow](RSKIP176/flow-chart-alternative.png)
+
+
+### Federation Address Derivation
+
+The address derivation process will be carried out by creating a custom redeem script that will make a push of a 32 bytes hash created from the provided derivation arguments. That data is then dropped.
+
+This new custom redeem scrtipt will have this structure::
+
+    scriptPubKey: OP_HASH160 <redeemScriptHash> OP_EQUAL
+    scriptSig: OP_0 <signatures> <redeemScript>
+    redeemScript: <derivationArgumentsHash> OP_DROP OP_M <publicKeys> OP_N OP_CHECKMULTISIG
+
+ This custom redeem script is executed in the same way to the standard one, allowing the same Federation members to sign the releases of these ad-hoc Federations.
+
```

Comment:
What is missing is the "Rationale" section where you explain the decisions that you made and other possible designs decisions that were discarded (i.e. the bridge calls the liquidity contract instead of the other way around)